repartitions/views.py

```diff
@@ -0,0 +1,196 @@
+import json
+
+from django.http import JsonResponse, HttpResponseBadRequest
```

Comment:
Why do you use `JsonResponse` instead of DRF's `Response`?
I understand for `HttpResponseBadRequest`, because DRF's equivalent is `ValidationError` which does not convey the same meaning